pymobiledevice3/service_connection.py

```diff
@@ -35,37 +35,41 @@ class ServiceConnection(object):
     def __init__(self, port, udid=None, logger=None):
         self.logger = logger or logging.getLogger(__name__)
         self.port = port
-        self.connect(udid)
+        self.udid = udid
+        self.connect()
 
-    def connect(self, udid=None):
+    def connect(self):
```

Comment:
Wouldn't it be prettier and SOLID to make this function a static method and to pass the socket as an argument to ServiceConnection?

Reply:
ok